src/library/searchqueryparser.cpp

```diff
@@ -48,13 +48,16 @@ SearchQueryParser::SearchQueryParser(QSqlDatabase& database)
     m_fieldToSqlColumns["rating"] << "rating";
     m_fieldToSqlColumns["location"] << "location";
     m_fieldToSqlColumns["datetime_added"] << "datetime_added";
-
+    m_fieldToSqlColumns["crate"] << "crates.name";
+    
     m_allFilters.append(m_textFilters);
+    m_allFilters.append("crate");
     m_allFilters.append(m_numericFilters);
     m_allFilters.append(m_specialFilters);
 
     m_fuzzyMatcher = QRegExp(QString("^~(%1)$").arg(m_allFilters.join("|")));
     m_textFilterMatcher = QRegExp(QString("^-?(%1):(.*)$").arg(m_textFilters.join("|")));
+    m_crateFilterMatcher = QRegExp(QString("^-?(crate):(.*)$"));
```

Comment:
Considering this again, I think it'd be better to reuse m_textFilterMatcher for crates. The way it is currently, both the m_crateFilterMatcher and m_textFilterMatcher regexes are evaluated for each QueryNode, which is wasteful for the regular text filters. Instead, you could add "crate" to m_textFilters and check if the first capture group in the regex is equal to "crate".